eth/db/chain.py

```diff
@@ -259,6 +261,15 @@ def _persist_block(
             for index, transaction_hash in enumerate(tx_hashes):
                 cls._add_transaction_to_canonical_chain(db, transaction_hash, header, index)
 
+            # post-shanghai, look for withdrawals
+            if hasattr(block, "withdrawals") and block.withdrawals not in (None, ()):
```

Comment:
**Note for refactoring:** This was made to work, re-think this logic a bit. Not just here, but this sort of logic appears in a few other places.